cluster.c

```diff
@@ -770,10 +757,21 @@ RRStatus ShardingInfoValidateShardGroup(RedisRaftCtx *rr, ShardGroup *new_sg)
         }
 
         for (int i = new_sg->slot_ranges[h].start_slot; i <= new_sg->slot_ranges[h].end_slot; i++) {
-            if (si->hash_slots_map[i] != 0) {
-                LOG_ERROR("Invalid shardgroup: hash slot already mapped: %u", i);
+            if (si->stable_slots_map[i] != 0) {
+                LOG_ERROR("Invalid shardgroup: hash slot already mapped as stable: %u", i);
                 return RR_ERROR;
             }
+            if (new_sg->slot_ranges[h].type == SLOTRANGE_TYPE_MIGRATING) {
+                if (si->migrating_slots_map[i] != 0) {
```

Comment:
```suggestion
                if (si->migrating_slots_map[i] != NULL) {
```
Do you think it is better to compare against NULL here and in other if checks as well (just for clarity)?

Reply:
yes, this is an artifact of the old slot map which mapped to an index.  it worked, but NULL is better